providers/implementations/serializers/serializer_rsa.c

```diff
@@ -109,10 +111,160 @@ int ossl_prov_print_rsa(BIO *out, RSA *rsa, int priv)
                 goto err;
         }
     }
+
+    switch (RSA_test_flags(rsa, RSA_FLAG_TYPE_MASK)) {
+    case RSA_FLAG_TYPE_RSA:
+        if (rsa_get0_pss_params_30(rsa) != NULL) {
+            if (ossl_prov_bio_printf(out, "(INVALID PSS PARAMETERS)\n") <= 0)
+                goto err;
+        }
+        break;
+    case RSA_FLAG_TYPE_RSASSAPSS:
+        {
+            const RSA_PSS_PARAMS_30 *pss_params = rsa_get0_pss_params_30(rsa);
+            int hashalg_nid = NID_sha1;
+            int maskgenalg_nid = NID_mgf1;
+            int maskgenhashalg_nid = NID_sha1;
+            int saltlen = 20;
+            int trailerfield = 1;
+
+            if (pss_params != NULL) {
+                hashalg_nid = rsa_pss_params_30_hashalg(pss_params);
+                maskgenalg_nid = rsa_pss_params_30_maskgenalg(pss_params);
+                maskgenhashalg_nid =
+                    rsa_pss_params_30_maskgenhashalg(pss_params);
+                saltlen = rsa_pss_params_30_saltlen(pss_params);
+                trailerfield = rsa_pss_params_30_trailerfield(pss_params);
+            }
+
+            if (ossl_prov_bio_printf(out, "PSS parameter restrictions:\n") <= 0)
+                goto err;
+            if (ossl_prov_bio_printf(out, "  Hash Algorithm: %s%s\n",
+                                     rsa_oaeppss_nid2name(hashalg_nid),
+                                     (hashalg_nid == NID_sha1
+                                      ? " (default)" : "")) <= 0)
+                goto err;
+            if (ossl_prov_bio_printf(out, "  Mask Algorithm: %s with %s%s\n",
+                                     rsa_mgf_nid2name(maskgenalg_nid),
+                                     rsa_oaeppss_nid2name(maskgenhashalg_nid),
+                                     (maskgenalg_nid == NID_mgf1
+                                      && maskgenhashalg_nid == NID_sha1
+                                      ? " (default)" : "")) <= 0)
+                goto err;
+            if (ossl_prov_bio_printf(out, "  Minimum Salt Length: %d%s\n",
+                                     saltlen,
+                                     (saltlen == 20 ? " (default)" : "")) <= 0)
+                goto err;
+            /*
+             * TODO(3.0) Should we show the ASN.1 trailerField value, or
```

Comment:
Looks like it is only ever this value - so it is probably not needed.

Reply:
*So far*, yes.